### PR TITLE
feat: add support for functional refs to various form components

### PR DESCRIPTION
**Pull request by joakbjerk**
Add test for `ref` as `function` for various input components

### CODE REVIEW

File: packages/dnb-eufemia/src/components/dropdown/__tests__/Dropdown.test.tsx [hunk 1]
```diff
@@ -1179,6 +1179,21 @@ describe('Dropdown component', () => {
     expect(ref.current instanceof HTMLButtonElement).toBe(true)
   })
 
+  it('gets valid buttonRef element when ref is function', () => {
+    const ref: React.MutableRefObject<HTMLButtonElement> =
+      React.createRef()
+
+    function refFuntion(instance: HTMLButtonElement) {
```
Comment (langz) [inline packages/dnb-eufemia/src/components/dropdown/__tests__/Dropdown.test.tsx:1186]:
```suggestion
    function refFunction(instance: HTMLButtonElement) {
```

File: packages/dnb-eufemia/src/components/dropdown/__tests__/Dropdown.test.tsx [hunk 2]
```diff
@@ -1179,6 +1179,21 @@ describe('Dropdown component', () => {
...
+      ref.current = instance
+    }
+
+    render(<Dropdown {...props} buttonRef={refFuntion} />)
```
Comment (langz) [inline packages/dnb-eufemia/src/components/dropdown/__tests__/Dropdown.test.tsx:1190]:
```suggestion
    render(<Dropdown {...props} buttonRef={refFunction} />)
```

File: packages/dnb-eufemia/src/components/dropdown/__tests__/Dropdown.test.tsx [hunk 3]
```diff
@@ -1194,6 +1209,21 @@ describe('Dropdown component', () => {
     expect(ref.current instanceof HTMLSpanElement).toBe(true)
   })
 
+  it('gets valid innerRef element when ref is function', () => {
+    const ref: React.MutableRefObject<HTMLButtonElement> =
+      React.createRef()
+
+    function refFuntion(instance: HTMLButtonElement) {
```
Comment (langz) [inline packages/dnb-eufemia/src/components/dropdown/__tests__/Dropdown.test.tsx:1216]:
```suggestion
    function refFunction(instance: HTMLButtonElement) {
```

File: packages/dnb-eufemia/src/components/dropdown/__tests__/Dropdown.test.tsx [hunk 4]
```diff
@@ -1194,6 +1209,21 @@ describe('Dropdown component', () => {
...
+      ref.current = instance
+    }
+
+    render(<Dropdown {...props} innerRef={refFuntion} />)
```
Comment (langz) [inline packages/dnb-eufemia/src/components/dropdown/__tests__/Dropdown.test.tsx:1220]:
```suggestion
    render(<Dropdown {...props} innerRef={refFunction} />)
```

**Review by langz — APPROVED** [review]
Tests go green, I go green ✅ 

Could there be any unwanted side effects of the `componentDidMount` changes/implementations 🤔 I hope/think not 🙏

**Comment by joakbjerk** [comment]
> Tests go green, I go green ✅
> 
> Could there be any unwanted side effects of the `componentDidMount` changes/implementations 🤔 I hope/think not 🙏

Hmm, I don't think so 😝 Just copied the way it was done in the `Button` component. But that does not mean that it haven't made caused any issues we have yet to pick up on 😅

**Review by tujoworker — APPROVED** [review]
Very good work 👏

**Pull request merged**